include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -439,6 +442,11 @@ void multiplexer::process_peek(boost::system::error_code error,
     next_layer_type::bytes_readable command(true);
     next_layer().io_control(command);
     std::size_t datagram_size = command.get();
+#ifdef __FreeBSD__
+    // FreeBSD's FIONREAD when asked of a UDP socket includes the UDP and IP headers which
+    // for IPv4 is 16 bytes and for IPv6 is 24 bytes
+    datagram_size -= remote_endpoint.address().is_v6() ? 24 : 16;
+#endif
```

Comment:
Yeah that took two days of tracing code to figure out. You just mentally assume that FIONREAD works, until you realise it was returning the wrong amount. BTW ASIO's socket::available() also gets this wrong, I reported it at https://github.com/chriskohlhoff/asio/issues/54.